relay-system/src/controller.rs

```diff
@@ -4,16 +4,45 @@ use std::time::Duration;
 use actix::actors::signal;
 use actix::fut;
 use actix::prelude::*;
-use futures::compat::Future01CompatExt;
 use futures01::future;
 use futures01::prelude::*;
+use once_cell::sync::OnceCell;
 use tokio::sync::watch;
 
 #[doc(inline)]
 pub use actix::actors::signal::{Signal, SignalType};
 
 use crate::compat;
 
+type ShutdownChannel = (
+    watch::Sender<Option<Shutdown>>,
+    watch::Receiver<Option<Shutdown>>,
+);
+
+/// Global [`ShutdownChannel`] for all services.
+static SHUTDOWN: OnceCell<ShutdownChannel> = OnceCell::new();
```

Comment:
This refactor made it possible to test the metrics aggregator without an actix runtime since we don't actually require shutdown logic. There are two changes:

 1. The public interface is almost identical, but the handle for shutdown signals is now an opaque type instead of a `watch::Receiver` directly. We still need an object instead of just an async function to retain cancellation safety for the future.
 2. To avoid talking to the actor, the shutdown channel is now moved to a lazy global. The static `Controller::shutdown_handle()` function already was a singleton, so having this global is not strictly worse. In a future refactor, this should be wired up using `Recipient` instead.